addons/sale_crm/views/sale_order_views.xml

```diff
@@ -15,7 +15,13 @@
         <field name="inherit_id" ref="sale.view_order_form"/>
         <field name="arch" type="xml">
             <xpath expr="//group[@name='technical']" position="inside">
-                <field name="opportunity_id" context="{'default_type': 'opportunity'}"/>
+                <field name="opportunity_id" context="{
+                    'default_type': 'opportunity',
+                    'default_partner_id': partner_id,
+                    'default_campaign_id': campaign_id,
+                    'default_medium_id': medium_id,
+                    'default_source_id': source_id,
```

Comment:
The sock lord is watching you :socks: :eyes: You can green the task once you have reorder the entries by alphabetic order. I have no other comment. I didn't get any issue during my tests.

Reply:
Ok thanks